sdk/core/core/inc/az_span.h

```diff
@@ -63,34 +71,62 @@ az_span_sub(az_span const span, size_t const begin, size_t const end) {
  * Returns `true` if a content of the @a span is equal to a content of the @b
  * span.
  */
+AZ_NODISCARD AZ_INLINE bool az_span_is_equal_content(az_span const a, az_span const b) {
+  return a._internal.length == b._internal.length
+      && memcmp(a._internal.begin, b._internal.begin, a._internal.length) == 0;
+}
+
+/**
+ * Returns `true` if a capacity of the @a span is equal to a capacity of the @b
+ * span.
+ */
+AZ_NODISCARD AZ_INLINE bool az_span_is_equal_capacity(az_span const a, az_span const b) {
```

Comment:
Do we need this?
Maybe we should just have a method that returns length and another method that returns capacity and the caller can compare them for equality. I thin it's good to have these methods no matter what. Then, if we use this method a lot, it can be private (not even a friend).

Reply:
no, this is actually a new method that I added to have `az_span_is_equal` addapted.  Becasue if we want to know if one az_span is equal to other, we are not only looking to length and content anymore, but we also need to check the capacity now. 
But if we still want to just check if the content is equal, that's why I introduced functions for equal_content and equal_capacity. So we cover the 3 comparation cases:
-  Contentent
-  Capacity
-  Content && Capacity

But yes, we need `az_span_capacity_get` and `az_span_length_get` and to use those everywhere instead of using span._internal.lenght 👍

Reply:
My instincts tell me that most of the cases I would need `az_span_is_equal(a, b)` that compares contents, and `az_span_get_capacity(a) == az_span_get_capacity(b)` when I need to compare capacities.
And when I write `az_span_is_equal_` and get two IntelliSense suggestions, `contents` and `capacity` that looks like a complication to me. I think there should be one meaning of equality, which is by contents disregarding capacity.